### PR TITLE
[harfbuzz] build hb-view

### LINKED ISSUE #34605: [harfbuzz] build the diagnostic tool hb-view for the optional harfbuzz feature cairo
### Is your feature request related to a problem? Please describe.

This is a follow-up issue to
https://github.com/microsoft/vcpkg/issues/34475

I would like to have the 'harfbuzz' utility tool 'hb-view' available for debugging purposes.

### Proposed solution

Hence change the file portfile.cmake for 'harfbuzz' accordingly.
'hb-view' will only be built if the optional harfbuzz features 'glib' and 'cairo' are available.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by a3emdot**
Fixes #34605.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~SHA512s are updated for each updated download~
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/harfbuzz/portfile.cmake [hunk 1]
```diff
@@ -100,6 +100,9 @@ configure_file("${CMAKE_CURRENT_LIST_DIR}/harfbuzzConfig.cmake.in"
 vcpkg_list(SET TOOL_NAMES)
 if("glib" IN_LIST FEATURES)
     vcpkg_list(APPEND TOOL_NAMES hb-subset hb-shape hb-ot-shape-closure hb-info)
+    if("cairo" IN_LIST FEATURES)
+        vcpkg_list(APPEND TOOL_NAMES hb-view)
+    endif()
 endif()
```
Comment (Cheney-W) [inline ports/harfbuzz/portfile.cmake:106]:
```suggestion
if("glib" IN_LIST FEATURES)
    vcpkg_list(APPEND TOOL_NAMES hb-subset hb-shape hb-ot-shape-closure hb-info)
endif()
if("cairo" IN_LIST FEATURES)
    vcpkg_list(APPEND TOOL_NAMES hb-view)
endif()
```
According to below comment:
`'hb-view' will only be built if the optional harfbuzz feature 'cairo' is available.`

So, you cannot add your changes under the condition where `if("glib" IN_LIST FEATURES)` is set to true.

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: ports/harfbuzz/portfile.cmake (hunk 1 above)
Comment (Cheney-W) [reply ports/harfbuzz/portfile.cmake:106]: It's strange. After applying my suggestions locally, I encountered an error when installing `harfbuzz[cairo]:x64-windows`:
```
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:36 (message):
  Couldn't find tool "hb-view"
```
So I'd like to clarify with you: is `hb-view` built only when the `cairo` feature is enabled?"

Comment (a3emdot) [reply ports/harfbuzz/portfile.cmake:106]: Sorry for being inaccurate in my comment, you'll need both 'glib' and 'cairo' for building 'hb-view', hence my suggested change still stands.
See
https://github.com/harfbuzz/harfbuzz/blob/fde1c3692a4a1e45ee48954fb486ae6c036a5687/util/meson.build#L23

Comment (a3emdot) [reply ports/harfbuzz/portfile.cmake:106]: I have updated this specific comment in
https://github.com/microsoft/vcpkg/issues/34605
and added 'glib' to it

**Review by Cheney-W — APPROVED** [review]

File: ports/harfbuzz/portfile.cmake (hunk 1 above)
Comment (Cheney-W) [reply ports/harfbuzz/portfile.cmake:106]: Thanks for your reply!

**Pull request merged**